src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -202,19 +275,350 @@ protected override ODataRequestMessageWrapper CreateRequestMessage(string method
             return this.CreateTopLevelRequest(method, requestUri, headers, httpStack, descriptor);
         }
 
+        /// <summary>
+        /// Gets the materializer to process the response.
+        /// </summary>
+        /// <param name="entityDescriptor">The entity descriptor whose response is getting materialized.</param>
+        /// <param name="responseInfo">Information about the response to be materialized.</param>
+        /// <returns>An instance of <see cref="MaterializeAtom"/>, that can be used to materialize the response.</returns>
         protected override MaterializeAtom GetMaterializer(EntityDescriptor entityDescriptor, ResponseInfo responseInfo)
         {
-            throw new NotImplementedException();
-        }
+            Debug.Assert(this.cachedResponse.Exception == null && this.materializerStateForDescriptor != null, "this.cachedResponse.Exception == null && this.descriptorForMaterializer != null");
+            
+            if (this.materializerStateForDescriptor.TryGetValue(entityDescriptor, out IMaterializerState value))
+            {
+                if (value is MaterializerEntry materializerEntry)
+                {
+                    return new MaterializeAtom(responseInfo, new[] { materializerEntry.Entry }, entityDescriptor.Entity.GetType(), materializerEntry.Format, base.MaterializerCache);
+                }
+            }

```

Comment:
You can get rid on the nested if
```suggestion
            if (this.materializerStateForDescriptor.TryGetValue(entityDescriptor, out IMaterializerState materializerState) && materializerState is MaterializerEntry materializerEntry)
            {
                return new MaterializeAtom(responseInfo, new[] { materializerEntry.Entry }, entityDescriptor.Entity.GetType(), materializerEntry.Format, base.MaterializerCache);
            }
```